packages/components/src/components/hds/text/types.ts

```diff
@@ -0,0 +1,58 @@
+export type HdsTextTags = Extract<keyof HTMLElementTagNameMap, 'span' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'div'>;
+
+export enum HdsTextColorValues {
+  Primary = 'primary',
+  Strong = 'strong',
+  Faint = 'faint',
+  Disabled = 'disabled',
+  HighContrast = 'high-contrast',
+  Action = 'action',
+  ActionHover = 'action-hover',
+  ActionActive = 'action-active',
+  Highlight = 'highlight',
+  HighlightOnSurface = 'highlight-on-surface',
+  HighlightHighContrast = 'highlight-high-contrast',
+  Success = 'success',
+  SuccessOnSurface = 'success-on-surface',
+  SuccessHighContrast = 'success-high-contrast',
+  Warning = 'warning',
+  WarningOnSurface = 'warning-on-surface',
+  WarningHighContrast = 'warning-high-contrast',
+  Critical = 'critical',
+  CriticalOnSurface = 'critical-on-surface',
+  CriticalHighContrast = 'critical-high-contrast',
+}
+export type HdsTextColors = `${HdsTextColorValues}`;
```

Comment:
I did not know this was a thing. I had always been setting the enums as the types on signatures. Does this make it so you don't have to actually use the enum or something?

Reply:
Yeah exactly! If I just use the enum, then someone specifying the value as a string like `"primary"` would get an error because they are not accessing it via the enum, even though it matches a valid enum value. This turns the enum into a string union of all the enum values, and allows consumers to use the string version `"primary"` or enum value `HdsTextColorValues.Primary`